lib/postgrex/extensions/binary.ex

```diff
@@ -304,6 +306,30 @@ defmodule Postgrex.Extensions.Binary do
     [flags|bin]
   end
 
+  defp encode_hstore(bin) do
+    keys_and_values = Enum.reduce bin, "", fn ({key, value}, acc) ->
+        [acc, encode_hstore_key(key), encode_hstore_value(value)]
+    end
+    :erlang.iolist_to_binary([<<Map.size(bin)::int32>> | keys_and_values])
+  end
+
+  defp encode_hstore_key(key) when not is_nil(key) do
+    encode_hstore_value key
+  end
+
+  defp encode_hstore_key(_key)  do
+    raise ArgumentError, message: "Hstore keys cannot be nil!"
+  end
+
+  defp encode_hstore_value(nil) do
+    <<-1::int32>>
+  end
+
+  defp encode_hstore_value(value) do
```

Comment:
I would add a `is_binary` clause here too (similar to how we should do it for keys).